src/poetry/utils/env.py

```diff
@@ -1839,6 +1842,46 @@ def ephemeral_environment(
         yield VirtualEnv(venv_dir, venv_dir)
 
 
+@contextmanager
+def build_environment(
+    poetry: P, env: Env | None = None, io: IO | None = None
+) -> Iterator[Env]:
+    """
+    If a build script is specified for the project, there could be additional build
+    time dependencies, eg: cython, setuptools etc. In these cases, we create an
+    ephemeral build environment with all requirements specified under
+    `build-system.requires` and return this. Otherwise, the given default project
+    environment is returned.
+    """
+    if not env or poetry.package.build_script:
+        with ephemeral_environment(executable=env.python if env else None) as venv:
+            overwrite = io and io.output.is_decorated() and not io.is_debug()
+            if io:
+                requires = map(
+                    lambda r: f"<c1>{r}</c1>", poetry.pyproject.build_system.requires
+                )
+                if not overwrite:
+                    io.write_line("")
+
+                io.overwrite(
+                    "<b>Preparing</b> build environment with build-system requirements"
```

Comment:
```suggestion
                    "<b>Preparing</b> build environment with build-system requirements: "
```

Reply:
THere is a whitespace in the next line and I prefer it to be  "build-system requirements a, b" instead of "build-system requirements: a, b" as it is more a sentence.